js/src/forum/states/NotificationListState.ts

```diff
@@ -0,0 +1,53 @@
+import PaginatedListState from '../../common/states/PaginatedListState';
+import Notification from '../../common/models/Notification';
+
+export default class NotificationListState extends PaginatedListState<Notification> {
+  constructor() {
+    super(10);
+
+    this.location = { page: 1 };
+  }
+
+  get type(): string {
+    return 'notifications';
+  }
+
+  public isLoading(): boolean {
+    return this.isInitialLoading() || this.isLoadingNext();
+  }
+
+  /**
+   * Load the next page of notification results.
+   */
+  load(): Promise<void> {
+    if (app.session.user.newNotificationCount()) {
+      this.pages = [];
```

Comment:
Shouldn't this reset location too?

Reply:
Do we want a method to reset location perhaps? Not sure. Don't like having `this.location = { page: 1 }` a lot tbh.